### PR TITLE
Random message

### PR DESCRIPTION
**Pull request by diegoro1**
Implemented a button that fetches quotes from a servlet and displays them randomly.

Check it out here: https://drodrigues-sps-spring21.uc.r.appspot.com 🙂

I will be changing the button's logic to disable the same index to be generated in a row. I thought about doing it recursively but I will probably be implementing this function in a less stack heavy way :) 

I would love any feedback!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/RandomQuoteServlet.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+/** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
```
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/RandomQuoteServlet.java]: Comment is out of date. Consider deleting it, or better yet updating it.

File: portfolio/src/main/webapp-local/src/components/Contacts/index.js [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+    // fetches data from '/random-string' servlet to replace message string
+    const setRandomString = async () => {
+        const response = await fetch('/random-string');
+        const quotes = await response.json();
+        const length = Object.keys(quotes).length;
+
+        const quote = randomQuoteGenerator(quotes, length);
+
+        // resets message to then place new quote
+        setMessage("");
```
Comment (perrycate) [inline portfolio/src/main/webapp-local/src/components/Contacts/index.js]: What's the advantage of setting it to empty first? Would just calling setMessage(quote) on its own not replace the first message?

File: portfolio/src/main/webapp-local/src/components/Contacts/index.js [hunk 3]
```diff
@@ -17,6 +38,12 @@ export default function Contacts() {
                     <a href="https://www.linkedin.com/in/rodrigues-diego"><LinkedInIcon className="icon" fontSize="large" style={{fill: "white"}}/></a>
                     <a href="https://github.com/diegoro1"><GitHubIcon className="icon" fontSize="large" style={{fill: "white"}}/></a>
                 </div>
+                <div className="button-container">
+                    <button className="random-button" onClick={setRandomString}>Random</button>
+                </div>
+                <div className="message-wrapper">
+                    <div className={(message === "")? "" : "message"}>{message}</div>
```
Comment (perrycate) [inline portfolio/src/main/webapp-local/src/components/Contacts/index.js:47]: What does this do?

ooooh moments after I wrote that I got it, I think. You're changing the class message if you don't have a message, so the div doesn't appear until you have something? If I'm right, consider adding an html comment explaining this. If I'm wrong, definitely correct me lol.

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -1 +1 @@
...=t[0]))}return e}var n={},o={1:0},u=[];function p(r){if(n[r])return n[r].exports;var t=n[r]={i:r,l:!1,exports:{}};return e[r].call(t.exports,t,t.exports,p),t.l=!0,t.exports}p.m=e,p.c=n,p.d=function(e,r,t){p.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},p.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},p.t=function(e,r){if(1&r&&(e=p(e)),8&r)return e;if(4&r&&"object"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(p.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:e}),2&r&&"string"!=typeof e)for(var n in e)p.d(t,n,function(r){return e[r]}.bind(null,n));return t},p.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return p.d(r,"a",r),r},p.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},p.p="/";var l=this.webpackJsonpwebapp=this.webpackJsonpwebapp||[],a=l.push.bind(l);l.push=r,l=l.slice();for(var f=0;f<l.length;f++)r(l[f]);var i=a;t()}([])</script><script src="/static/js/2.f319fc49.chunk.js"></script><script src="/static/js/main.b9f95483.chunk.js"></script></body></html>
```
Comment (perrycate) [inline portfolio/src/main/webapp/index.html:1]: Just for my own understanding: Is this the minified output of the build command that you've copy-pasted?

**Review by perrycate — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/RandomQuoteServlet.java (hunk 1 above)
Comment (diegoro1) [reply portfolio/src/main/java/com/google/sps/servlets/RandomQuoteServlet.java]: Updating now!

File: portfolio/src/main/webapp-local/src/components/Contacts/index.js (hunk 2 above)
Comment (diegoro1) [reply portfolio/src/main/webapp-local/src/components/Contacts/index.js]: Your logic is absolutely right, but for some reason the quotes were stacking on each other. I will delete it again and see if it were a mistake on maybe another line of my code that is no longer there

File: portfolio/src/main/webapp-local/src/components/Contacts/index.js (hunk 3 above)
Comment (diegoro1) [reply portfolio/src/main/webapp-local/src/components/Contacts/index.js:47]: You are correct! I will add a comment :)

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (diegoro1) [reply portfolio/src/main/webapp/index.html:1]: Also correct! That's the index.html generated by running `npm run build`

Comment (perrycate) [reply portfolio/src/main/webapp/index.html:1]: Got it. In that case you can probably add it to .gitignore along with the other stuff if you want.

**Pull request merged**